server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -1860,9 +1860,14 @@ private CompactionStats _majorCompact(MajorCompactionReason reason) throws IOExc
 
       if (inputFiles.isEmpty()) {
         if (reason == MajorCompactionReason.USER) {
-          // no work to do
-          lastCompactID = compactionId.getFirst();
-          updateCompactionID = true;
+          if (compactionId.getSecond().getIterators().isEmpty()) {
+            log.debug("No-op major compaction by USER on 0 input files because no iterators present.");
+            lastCompactID = compactionId.getFirst();
+            updateCompactionID = true;
+          } else {
+            log.debug("Major compaction by USER on 0 input files with iterators.");
+            filesToCompact = new HashMap<FileRef,DataFileValue>();
```

Comment:
Hey Josh, question on open-source contributing.  There are many, many spots throughout the Tablet class where we can delete the contents of the diamond.  However, deleting them all isn't relevant to the goal of this patch: making iterators run on empty user compactions. If I deleted them all, then reviewers may find the patch confusing.

Is opening a new ticket and deleting all the diamonds the thing to do?  It seems kinda silly to use a ticket for that.  Actually if we need to open a ticket for every little issue, then no one would ever fix (for example) a typo in documentation because the overhead of a ticket is overkill.  How does it work?